src/Parsers/Access/ParserCreateQuotaQuery.cpp

```diff
@@ -107,19 +108,27 @@ namespace
         });
     }
 
+    template <typename T>
+    T fieldToNumber(const Field & f)
+    {
+        if (f.getType() == Field::Types::String)
+            return parseWithSizeSuffix<T>(f.get<const String &>());
```

Comment:
How do you handle error input here? Please add some negative-case tests, like:
- `query_inserts = '1!'`
- `query_inserts = '1test'`
- etc

Reply:
Exception will be throwed, well it's better not to throw exception in parser.

Reply:
IMO it is not that bad... CH already throws exceptions from parser code. BUT still having an error-case test would be really nice.
```
src/Parsers$ rgrep --include 'Parser*.cpp' -Han throw
ParserCreateQuery.cpp:687:            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Multiple primary keys are not allowed.");
ParserSetQuery.cpp:260:            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Parameter name cannot be empty");
ParserInsertQuery.cpp:201:            throw Exception(ErrorCodes::SYNTAX_ERROR,
ParserInsertQuery.cpp:231:            throw Exception(ErrorCodes::SYNTAX_ERROR, "You have excessive ';' symbol before data for INSERT.\n"
ParserTablesInSelectQuery.cpp:193:                throw Exception(ErrorCodes::SYNTAX_ERROR, "You must not specify ANY or ALL for CROSS JOIN.");
ParserTablesInSelectQuery.cpp:197:                throw Exception(ErrorCodes::SYNTAX_ERROR, "SEMI|ANTI JOIN should be LEFT or RIGHT.");
ParserExternalDDLQuery.cpp:65:                throw Exception(ErrorCodes::MYSQL_SYNTAX_ERROR, "Cannot parse MySQL alter query.");
ParserExternalDDLQuery.cpp:68:                throw Exception(ErrorCodes::MYSQL_SYNTAX_ERROR, "Cannot parse MySQL rename query.");
ParserExternalDDLQuery.cpp:71:                throw Exception(ErrorCodes::MYSQL_SYNTAX_ERROR, "Cannot parse MySQL drop query.");
ParserExternalDDLQuery.cpp:74:                throw Exception(ErrorCodes::MYSQL_SYNTAX_ERROR, "Cannot parse MySQL create query.");
ParserSelectQuery.cpp:327:            throw Exception(ErrorCodes::LIMIT_BY_WITH_TIES_IS_NOT_SUPPORTED, "Can not use WITH TIES alongside LIMIT BY/DISTINCT ON");
ParserSelectQuery.cpp:335:                throw Exception(ErrorCodes::LIMIT_BY_WITH_TIES_IS_NOT_SUPPORTED, "Can not use WITH TIES alongside LIMIT BY/DISTINCT ON");
ParserSelectQuery.cpp:338:                throw Exception(ErrorCodes::SYNTAX_ERROR, "Can not use DISTINCT ON alongside LIMIT BY");
ParserSelectQuery.cpp:350:            throw Exception(ErrorCodes::TOP_AND_LIMIT_TOGETHER, "Can not use TOP and LIMIT together");
ParserSelectQuery.cpp:363:                throw Exception(ErrorCodes::ROW_AND_ROWS_TOGETHER, "Can not use ROW and ROWS together");
ParserSelectQuery.cpp:375:                throw Exception(ErrorCodes::OFFSET_FETCH_WITHOUT_ORDER_BY, "Can not use OFFSET FETCH clause without ORDER BY");
ParserSelectQuery.cpp:380:                    throw Exception(ErrorCodes::FIRST_AND_NEXT_TOGETHER, "Can not use FIRST and NEXT together");
ParserSelectQuery.cpp:391:                    throw Exception(ErrorCodes::ROW_AND_ROWS_TOGETHER, "Can not use ROW and ROWS together");
ParserSelectQuery.cpp:455:        throw Exception(ErrorCodes::WITH_TIES_WITHOUT_ORDER_BY, "Can not use WITH TIES without ORDER BY");
Kusto/ParserKQLOperators.cpp:23:        throw Exception(ErrorCodes::SYNTAX_ERROR, "Syntax error near {}", kql_op);
Kusto/ParserKQLOperators.cpp:58:        throw Exception(ErrorCodes::SYNTAX_ERROR, "Syntax error near {}", kql_op);
Kusto/ParserKQLOperators.cpp:118:        throw Exception(ErrorCodes::SYNTAX_ERROR, "Syntax error near {}", kql_op);
Kusto/ParserKQLOperators.cpp:138:                throw Exception(ErrorCodes::SYNTAX_ERROR, "Invalid negative operator");
Access/ParserCreateUserQuery.cpp:189:                        throw Exception(ErrorCodes::OPENSSL_ERROR, "Cannot generate salt for password. OpenSSL {}", buf);
Access/ParserCreateUserQuery.cpp:203:                    ///if USE_SSL is not defined, Exception thrown later
Access/ParserGrantQuery.cpp:159:    void throwIfNotGrantable(AccessRightsElements & elements)
Access/ParserGrantQuery.cpp:171:                throw Exception(ErrorCodes::INVALID_GRANT, "{} cannot be granted on the column level", old_flags.toString());
Access/ParserGrantQuery.cpp:173:                throw Exception(ErrorCodes::INVALID_GRANT, "{} cannot be granted on the table level", old_flags.toString());
Access/ParserGrantQuery.cpp:175:                throw Exception(ErrorCodes::INVALID_GRANT, "{} cannot be granted on the database level", old_flags.toString());
Access/ParserGrantQuery.cpp:177:                throw Exception(ErrorCodes::INVALID_GRANT, "{} cannot be granted", old_flags.toString());
Access/ParserGrantQuery.cpp:284:        throw Exception(ErrorCodes::SYNTAX_ERROR, "GRANT OPTION should be specified for access types");
Access/ParserGrantQuery.cpp:286:        throw Exception(ErrorCodes::SYNTAX_ERROR, "ADMIN OPTION should be specified for roles");
Access/ParserGrantQuery.cpp:310:            throwIfNotGrantable(elements);
Access/ParserShowCreateAccessEntityQuery.cpp:153:            throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Type {} is not implemented in SHOW CREATE query", toString(type));
Access/ParserCreateQuotaQuery.cpp:74:            throw Exception(ErrorCodes::SYNTAX_ERROR, "Quota cannot be keyed by '{}'. Expected one of the following identifiers: {}", name, all_types_str);
```